*** tclXprofile.c.orig	Wed Nov 20 17:53:19 1996
--- tclXprofile.c	Sat Nov 30 13:25:43 1996
***************
*** 525,538 ****
  {
      Interp *iPtr = (Interp *) interp;
      profInfo_t *infoPtr = (profInfo_t *) clientData;
-     Tcl_HashEntry *hPtr;
      Command *cmdPtr;
!     
      if (infoPtr->currentCmdPtr != NULL)
          panic (PROF_PANIC, 5);
  
      hPtr = Tcl_FindHashEntry (&iPtr->commandTable, argv [0]);
      cmdPtr = (Command *) Tcl_GetHashValue (hPtr);
  
      if ((cmdPtr->proc != cmdProc) || (cmdPtr->clientData != cmdClientData))
          panic (PROF_PANIC, 6);
--- 525,548 ----
  {
      Interp *iPtr = (Interp *) interp;
      profInfo_t *infoPtr = (profInfo_t *) clientData;
      Command *cmdPtr;
! #ifdef ITCL_NAMESPACES      
!     Tcl_Command cmd;
! #else
!     Tcl_HashEntry *hPtr;
! #endif    
! 
      if (infoPtr->currentCmdPtr != NULL)
          panic (PROF_PANIC, 5);
  
+ #ifdef ITCL_NAMESPACES      
+     if (Itcl_FindCommand((Tcl_Interp*)iPtr, argv [0], 0, &cmd) != TCL_OK)
+         panic (PROF_PANIC, 56);
+     cmdPtr = (Command*)cmd;
+ #else
      hPtr = Tcl_FindHashEntry (&iPtr->commandTable, argv [0]);
      cmdPtr = (Command *) Tcl_GetHashValue (hPtr);
+ #endif
  
      if ((cmdPtr->proc != cmdProc) || (cmdPtr->clientData != cmdClientData))
          panic (PROF_PANIC, 6);
